Make example functions in Challenge seed unique

ChadKreutzer opened the pull request:
<!-- FreeCodeCamp Pull Request Template -->

<!-- IMPORTANT Please review https://github.com/FreeCodeCamp/FreeCodeCamp/blob/staging/CONTRIBUTING.md for detailed contributing guidelines -->

<!-- Help with PRs can be found at https://gitter.im/FreeCodeCamp/HelpContributors -->

<!-- Make sure that your PR is not a duplicate -->
#### Pre-Submission Checklist

<!-- Go over all points below, and after creating the PR, tick all the checkboxes that apply. -->

<!-- All points should be verified, otherwise, read the CONTRIBUTING guidelines from above-->

<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] Your pull request targets the `staging` branch of FreeCodeCamp.
- [X] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [X] You have only one commit (if not, [squash](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/git-rebase#squashing-multiple-commits-into-one) them into one commit).
- [x] All new and existing tests pass the command `npm run test-challenges`. Use `git commit --amend` to amend any fixes.
#### Type of Change

<!-- What type of change does your code introduce? After creating the PR, tick the checkboxes that apply. -->
- [x] Small bug fix (non-breaking change which fixes an issue)
#### Checklist:

<!-- Go over all points below, and after creating the PR, tick the checkboxes that apply. -->

<!-- If you're unsure about any of these, don't hesitate to ask in the Help Contributors room linked above. We're here to help! -->
- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #9456 
#### Description

<!-- Describe your changes in detail -->

…vaScript with Functions, Passing Values to Functions with Arguments, and Return a Value from a Function with Return.

ChadKreutzer committed:
improve challengeSeed for clarity and uniformity in Write Reusable JavaScript with Functions, Passing Values to Functions with Arguments, and Return a Value from a Function with Return

removed comma

Changed to reflect ourUniqueName()/uniqueName() pattern throughout

removed debugging artifact

change ourFunction(1, 9); to ourRandomRange(1, 9);

Remove second function calls where appropriate.

ChadKreutzer commented:
I'm confused by camperbot. I could have sworn the instructions said **TO** put "closes #XXXX" in the commit. In fact I amended my commit to do so.

atjonathan commented on seed/challenges/01-front-end-development-certification/basic-javascript.json:
```diff
@@ -2157,14 +2161,18 @@
+        "// Only change code above this line",
+        "",
+        "functionWithArgs(10, 5); // Should Output 10",
+        "",
```
Remove the comma at the end of this line.

ChadKreutzer commented:
Looking at the details of the failing check, it doesn't seem to be anything I did that is failing, so I am confused by that too.

raisedadead commented on seed/challenges/01-front-end-development-certification/basic-javascript.json:
```diff
@@ -2419,6 +2427,10 @@
         "",
         "// Only change code below this line",
         "",
+        "",
```
@ChadKreutzer Can you please guide what is this change refering to?

ChadKreutzer replied:
removing a comma at the end of the entry that I had accidentally left there.

raisedadead replied:
@ChadKreutzer Can you please guide what is this change referring to?

raisedadead replied:
Apologies please check the comment on the next diff.

raisedadead commented:
Also it would be great if you could make the changes to Challenges mentioned in the original issue.
Refer : https://github.com/FreeCodeCamp/FreeCodeCamp/issues/9456#issuecomment-229705986

Thanks a lot for this PR.

P.S : Your commit still contains the issue no.

ChadKreutzer commented:
@raisedadead Fixed the commit message. I'm a little confused by your other issue, since I'm relatively certain that is exactly what I did. Could you elaborate?

raisedadead commented:
@ChadKreutzer from the original issue:

>Yeah, it seems that `ourFunction` for the example and a unique function name for the user is also used in [Use Conditional Logic with If Statements](https://www.freecodecamp.com/challenges/use-conditional-logic-with-if-statements).
>
>Maybe in all such cases it would be clearer to use the unique function name prefixed with `our` for the example, and without it for the user's function, i.e. `ourReusableFunction` / `reusableFunction` for this challenge and `ourTrueOrFalse` / `trueOrFalse` for the Conditional Logic challenge linked above.
>
>Or maybe that camel case variation is confusing :)
>
>Edit: there's at least one other challenge that asks the user to create a new function with the same name as the example function, [Passing Values to Functions with Arguments](https://www.freecodecamp.com/challenges/passing-values-to-functions-with-arguments), which could definitely benefit from this kind of change!


Could you update these names at all such places, similarly?

ChadKreutzer commented:
Okay. I see what your meaning now. I will get that fixed. :-)

raisedadead commented on package.json:
```diff
@@ -135,6 +135,7 @@
     "yargs": "^4.1.0"
   },
   "devDependencies": {
+    "babel-cli": "^6.10.1",
```
Remove this

ChadKreutzer commented:
I had changed the ones mentioned, but I had not followed the suggested pattern and instead had followed the pattern of "ourFuncton()/uniqueFunction()" I have now changed all the applicable challenges. (I did not change the ones that followed the ourX/myX pattern, because they seemed to make sense and be consistent within their series.)

raisedadead commented:
@ChadKreutzer could you hit us up in the [Contributors Chat room](https://gitter.im/FreeCodeCamp/HelpContributors).

raisedadead commented:
@erictleung I am supposing you might have a look as well.

erictleung commented on seed/challenges/01-front-end-development-certification/basic-javascript.json:
```diff
@@ -2419,6 +2427,10 @@
+        "",
+        "// Only change code above this line",
+        "",
+        "timesFive(5);",
```
@ChadKreutzer @raisedadead wouldn't this create an error when you initialize the challenge because there is no function defined yet called `timesFive`? I don't think these three lines are necessary. Unless there is another reason why this is here.

ChadKreutzer replied:
I haven't received any errors. The reason I put those there, as well as the `// Only change code above this line` comment was to keep a uniform appearance and experience with other challenges that have similar setups. Especially the function call at the bottom: just about every challenge follows that pattern.

raisedadead replied:
IMHO @ChadKreutzer this function needs to be typed in by the camper. I think its best we remove this. And similarly else where.

ChadKreutzer replied:
All they need to type in is the function definition, all of the challenges pass without the function call on the bottom at all. Although they do not pass if that function is not defined at all. I tend to look on the bottom function call in the lessons as just a helper (in fact, I usually would wrap it in a console log while I was working on the lesson and never had any issues). In the case of these where the camper is asked to create a function of that name, the fact that it won't pass without the function defined properly is an added little data point for the camper.

erictleung replied:
@ChadKreutzer correct, the function call isn't needed to pass the challenge. However, the code as you have it will prevent the camper from knowing what they need to do and they will have to delete that line anyways. 

![image](https://cloud.githubusercontent.com/assets/2754821/16537799/800c06a6-3fc2-11e6-842f-caf59ffc1933.png)

ChadKreutzer replied:
I think we might be talking a little bit at cross purposes. You are absolutely right that until the camper defines timesFive, it'll throw an error. the way I was looking at it was the function call to help with testing and debugging that is built into most of the later challenges and thus would not need to be deleted at all. That being said, I can see how it might be confusing to someone inexperienced with functions. I'll go ahead and remove them.

erictleung commented on seed/challenges/01-front-end-development-certification/basic-javascript.json:
```diff
@@ -2157,13 +2161,17 @@
+        "",
+        "// Only change code above this line",
+        "",
+        "functionWithArgs(10, 5); // Should Output 10",
```
@ChadKreutzer @raisedadead same with the comment below. This function call `functionWithArgs(10, 5)` would throw an error, I believe, because this function is not defined yet. These four lines appear to be unnecessary.

erictleung commented on seed/challenges/01-front-end-development-certification/basic-javascript.json:
```diff
@@ -2077,14 +2077,18 @@
+        "",
+        "// Only change code above this line",
+        "",
+        "reusableFunction();",
```
@ChadKreutzer @raisedadead again, possible error because no function defined as `reusableFunction()` yet.

erictleung commented:
@ChadKreutzer see comments above. Also, [this line](https://github.com/FreeCodeCamp/FreeCodeCamp/pull/9485/files#diff-5fbffcdf48f7976dd3b0216c387eebc2R5112) needs to be changed to `ourRandomRange(1, 9)`.

Please make or address the comments above and then we can merge your PR. Thanks for the PR nonetheless!

raisedadead commented:
Thanks for the QA Eric, I was yet to do the same. Just got back from work.

ChadKreutzer commented:
I did change the ourRandomRange line that I missed, though.

ChadKreutzer commented:
@erictleung @raisedadead Alright. hopefully this is good :smiley:

raisedadead commented:
@ChadKreutzer thanks this **LGTM**, 

@atjonathan @erictleung should be good I guess?

atjonathan commented:
### **LGTM :+1:**

erictleung commented:
@ChadKreutzer LGTM! 👍  Thanks for your contribution to FreeCodeCamp! ✨ 🎉 ✨
